src/main/java/jenkinsci/plugins/influxdb/generators/CustomDataMapPointGenerator.java

```diff
@@ -28,10 +32,17 @@ public boolean hasReport() {
     public Point[] generate() {
         List<Point> customPoints = new ArrayList<Point>();
         Set<String> customKeys = customDataMap.keySet();
+
         for (String key : customKeys) {
-            Point point = buildPoint(measurementName(key), customPrefix, build)
-                    .fields(customDataMap.get(key))
-                    .build();
+            Point.Builder pointBuilder = buildPoint(measurementName(key), customPrefix, build)
+                    .fields(customDataMap.get(key));
+
+            if (customDataMapTags != null && customDataMapTags.size() > 0) {
+                pointBuilder = pointBuilder.tag(customDataMapTags);
```

Comment:
Currently, this is writing the same tags for all measurements.
I would suggest to do it similar to line 38 and allow different tags per measurement.

Usage would be then like
```
def myCustomTags = [:]
myTags1 = [buildResult: currentBuild.result, NODE_LABELS: env.NODE_LABELS]
myTags2 = [testTag: 'testTag']
myCustomTags["series1"] = myTags1
myCustomTags["series2"] = myTags2
```

This is similar to usage of `customDataMap`: https://github.com/jenkinsci/influxdb-plugin/blob/master/src/main/java/jenkinsci/plugins/influxdb/InfluxDbPublisher.java#L95

Above behavior fits better to `customData`: https://github.com/jenkinsci/influxdb-plugin/blob/master/src/main/java/jenkinsci/plugins/influxdb/InfluxDbPublisher.java#L72

Reply:
Thanks for the feedback. I agree that would better mirror how cutomDataMap is used. I've updated the code to reflect the changes.